src/nvim/eval/typval_encode.c.h

```diff
@@ -351,8 +351,16 @@ static int TYPVAL_ENCODE_CONVERT_ONE_VALUE(
     break;
   case VAR_PARTIAL: {
     partial_T *const pt = tv->vval.v_partial;
-    (void)pt;
-    TYPVAL_ENCODE_CONV_FUNC_START(tv, (pt == NULL ? NULL : partial_name(pt)));
+    char *const fname = pt == NULL ? NULL : partial_name(pt);
+    char *ffname = NULL;
+    if (fname != NULL && pt->pt_name == NULL && ASCII_ISUPPER(fname[0])) {
+      size_t len = strlen(fname) + 3;
+      ffname = xmalloc(len);
+      snprintf(ffname, len, "g:%s", fname);
+    }
+    (void)fname;
+    (void)ffname;
+    TYPVAL_ENCODE_CONV_FUNC_START(tv, ffname != NULL ? ffname : fname);
```

Comment:
I don't think prepending `g:` here is correct. In the `TYPVAL_ENCODE_CONV_FUNC_START()` of lua/converter.c and api/private/converter.c, `fun` is passed to `find_func()` to look it up inside `func_hashtab`, which doesn't use `g:` prefix.

Instead prepending `g:` should only be done in the first `TYPVAL_ENCODE_CONV_FUNC_START()` of eval/encode.c.

Reply:
Or, maybe it's better to pass a static string prefix (`""` or `"g:"` depending on the condition here) as an extra argument to `TYPVAL_ENCODE_CONV_FUNC_START()`, and let `TYPVAL_ENCODE_CONV_FUNC_START()` do the actual prepending if that's necessary.

Reply:
I'm trying this in #37031.